FreeRTOS/Test/CMock/queue/generic/queue_receive_blocking_utest.c

```diff
@@ -0,0 +1,614 @@
+
+    uint32_t checkVal = INVALID_UINT32;
+
+    TEST_ASSERT_EQUAL( pdFALSE, xQueuePeek( xQueue, &checkVal, TICKS_TO_WAIT ) );
+
+    TEST_ASSERT_EQUAL( TICKS_TO_WAIT, td_task_getYieldCount() );
+
+    TEST_ASSERT_EQUAL( NUM_CALLS_TO_INTERCEPT + 1, td_task_getCount_YieldFromTaskResumeAll() );
+
+    TEST_ASSERT_EQUAL( NUM_CALLS_TO_INTERCEPT - 1, td_task_getCount_vPortYieldWithinAPI() );
+
+    TEST_ASSERT_EQUAL( 1, td_task_getCount_vTaskMissedYield() );
+
+    TEST_ASSERT_EQUAL( INVALID_UINT32, checkVal );
+
+    vQueueDelete( xQueue );
+}
+
+/**
+ *  @brief Test a blocking call to xQueueReceive with a locked queue.
+ *  @details Test a blocking call to xQueueReceive with a locked queue with a lower
+ *  priority task in the queue WaitingToReceiveFrom event list.
+ *  @coverage xQueueReceive
+ */
+void test_macro_xQueueReceive_blocking_success_locked_low_prio_pending( void )
+{
+    QueueHandle_t xQueue = xQueueCreate( 1, sizeof( uint32_t ) );
+
+    /* Export for callbacks */
+    xQueueHandleStatic = xQueue;
+
+    xTaskCheckForTimeOut_Stub( &xQueueReceive_xTaskCheckForTimeOutCB );
+    xTaskResumeAll_Stub( &xQueueReceive_xTaskResumeAllCallback );
+
+    td_task_setFakeTaskPriority( DEFAULT_PRIORITY - 1 );
+
+    td_task_addFakeTaskWaitingToReceiveFromQueue( xQueue );
+
+    uint32_t checkVal = INVALID_UINT32;
+
+    TEST_ASSERT_EQUAL( pdTRUE, xQueueReceive( xQueue, &checkVal, TICKS_TO_WAIT ) );
+
+    TEST_ASSERT_EQUAL( NUM_CALLS_TO_INTERCEPT, td_task_getYieldCount() );
+
+    TEST_ASSERT_EQUAL( NUM_CALLS_TO_INTERCEPT, td_task_getCount_vPortYieldWithinAPI() );
+
+    TEST_ASSERT_EQUAL( getLastMonotonicTestValue(), checkVal );
+
+    vQueueDelete( xQueue );
+}
+
+/**
+ *  @brief Test a blocking call to xQueuePeek with a locked queue.
+ *  @details Test a blocking call to xQueuePeek with a locked queue with a lower
+ *  priority task in the queue WaitingToReceiveFrom event list.
+ *  @coverage xQueuePeek
+ */
+void test_macro_xQueuePeek_blocking_success_locked_low_prio_pending( void )
+{
+    QueueHandle_t xQueue = xQueueCreate( 1, sizeof( uint32_t ) );
+
+    /* Export for callbacks */
+    xQueueHandleStatic = xQueue;
+
+    xTaskCheckForTimeOut_Stub( &xQueueReceive_xTaskCheckForTimeOutCB );
+    xTaskResumeAll_Stub( &xQueueReceive_xTaskResumeAllCallback );
+
+    td_task_setFakeTaskPriority( DEFAULT_PRIORITY - 1 );
+
+    td_task_addFakeTaskWaitingToReceiveFromQueue( xQueue );
+
+    uint32_t checkVal = INVALID_UINT32;
+
+    TEST_ASSERT_EQUAL( pdTRUE, xQueuePeek( xQueue, &checkVal, TICKS_TO_WAIT ) );
+
+    TEST_ASSERT_EQUAL( NUM_CALLS_TO_INTERCEPT, td_task_getYieldCount() );
+
+    TEST_ASSERT_EQUAL( NUM_CALLS_TO_INTERCEPT, td_task_getCount_vPortYieldWithinAPI() );
+
+    TEST_ASSERT_EQUAL( getLastMonotonicTestValue(), checkVal );
+
+    vQueueDelete( xQueue );
+}
+
+/**
+ *  @brief Test xQueuePeek with taskSCHEDULER_SUSPENDED and timeout=10
+ *  @details This should cause xQueuePeek to configASSERT becuase it would
```

Comment:
Is this assert as part of the code or from the test stubs?

Reply:
The assertion is from xQueuePeek
```
configASSERT( !( ( xTaskGetSchedulerState() == taskSCHEDULER_SUSPENDED ) && ( xTicksToWait != 0 ) ) );
```